### PR TITLE
New IntArray capabilities

### PR DESCRIPTION
**Pull request by victorapm**
This PR adds:
- `hypre_IntArraySeparateByValue`: function for creating new arrays after segregating indices of an input array by value. (to be used in MGR)
- New `hypre_IntArrayArray` type to denote an array of `hypre_IntArray`, just like `hypre_BoxArrayArray` does for `hypre_BoxArray`.

### CODE REVIEW

File: src/utilities/int_array.c [hunk 1]
```diff
@@ -537,3 +537,191 @@ hypre_IntArrayNegate( hypre_IntArray *v )
...
+ *   - values = {-1, 1}
+ *   - num_values = 2
+ *
+ *   This function computes:
+ *   - w[0] = {0, 1, 3, 4, 7}
+ *   - w[1] = {2, 5, 6}
+ *--------------------------------------------------------------------------*/
+
+HYPRE_Int
+hypre_IntArraySegregateByValue( HYPRE_Int             num_values,
```
Comment (rfalgout) [inline src/utilities/int_array.c]: Not sure about the name "Segregate".  Too many negative connotations.  Maybe just "Separate"?

**Review by rfalgout — APPROVED** [review]
Looks good to me, except for the name.  Thanks!

File: src/utilities/int_array.c (hunk 1 above)
Comment (victorapm) [reply src/utilities/int_array.c]: Yes, let me change it, thanks!

Comment (victorapm) [reply src/utilities/int_array.c]: Done in [c44f000](https://github.com/hypre-space/hypre/pull/1036/commits/c44f000f526305a2bc30247b697b07ded4e02aed)

**Pull request merged**